### PR TITLE
RenderTexturePool and FilterSystem changes

### PR DESCRIPTION
**Pull request by ivanpopelyshev**
WIP, here'll be demos soon, explaining everything.

### CODE REVIEW

**Comment by bigtimebuddy** [comment]
Demo? Also is this done?

**Comment by ivanpopelyshev** [comment]
Demo: https://www.pixiplayground.com/#/edit/4rskiJ8lb8k6437WmSwKy

OK, while making last-minute changes to #5771 I actually broke that demo :(

The problems are 
1. RenderTexturePool does not depend on renderer anymore, so it cant look at current renderTexture, and `getFilterTexture` needs RT as an argument.
2. getFilterTexture should have the same interface in FilterSystem and in RenderTexturePool to actually swap them easily in filters.

Another idea is that Texture should have `get resolution()`, to simplify `texture.baseTexture.resolution` , we use it often in places that aren't performance critical.

File: packages/core/src/textures/Texture.js [hunk 1]
```diff
@@ -495,6 +495,16 @@ export default class Texture extends EventEmitter
         return null;
     }
 
+    /**
+     * Returns resolution of baseTexture
+     *
+     * @readonly
+     */
+    get resolution()
```
Comment (GoodBoyDigital) [inline packages/core/src/textures/Texture.js]: nice!

File: packages/core/src/renderTexture/RenderTexturePool.js [hunk 2]
```diff
@@ -93,18 +96,20 @@ export default class RenderTexturePool
...
+     * `getFilterTexture(input, 0.5)` or `getFilterTexture(0.5, input)`
      *
-     * @param {number} [resolution] resolution, by default its the same as in current renderTexture
+     * @param {PIXI.RenderTexture} input renderTexture from which size and resolution will be copied
+     * @param {number} [resolution] override resolution of the renderTexture
+     *  It overrides, it does not multiply
      * @returns {PIXI.RenderTexture}
      */
-    getFilterTexture(resolution)
+    getFilterTexture(input, resolution)
```
Comment (GoodBoyDigital) [inline packages/core/src/renderTexture/RenderTexturePool.js]: I think we should rename input to something that makes it easy to know its a renderTexture `targetRenderTexture` ?

Comment (ivanpopelyshev) [reply packages/core/src/renderTexture/RenderTexturePool.js]: its `input` in filters. Name is the same as in https://github.com/pixijs/pixi.js/blob/dev/packages/core/src/filters/Filter.js#L215 . Its supposed to be called from apply()

**Pull request merged**